libs/editor/libs/editor-common/assets/ZSSRichTextEditor.js

```diff
@@ -924,24 +924,42 @@ ZSSEditor.replaceLocalImageWithRemoteImage = function(imageNodeIdentifier, remot
         ZSSEditor.markImageUploadDone(imageNodeIdentifier);
         var joinedArguments = ZSSEditor.getJoinedFocusedFieldIdAndCaretArguments();
         ZSSEditor.callback("callback-input", joinedArguments);
-
+        image.onerror = null;
+        image.classList.add("image-loaded");
     }
 
     image.onerror = function () {
-        // Even on an error, we swap the image for the time being.  This is because private
-        // blogs are currently failing to download images due to access privilege issues.
-        //
-        imageNode.attr('src', image.src);
```

Comment:
> What would be the best solution to mark it "unswapped yet", add a class to the img?

I think a good solution would be adding a `remoteUrl` attribute, set to the remote URL once we have it. Then we add an extra step to [`ZSSEditor.removeVisualFormatting()`](https://github.com/wordpress-mobile/WordPress-Editor-Android/blob/develop/libs/editor-common/assets/ZSSRichTextEditor.js#L2071-L2078) to find any image tags with that attribute and swap out the `src` for `remoteUrl`. That should cover the case where the image never loads before publishing.